File: src/main/java/net/hectus/neobb/turn/default_game/flower/TPinkTulip.java
package net.hectus.neobb.turn.default_game.flower;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.SupernaturalClazz;
import org.bukkit.block.Block;

import java.util.List;

public class TPinkTulip extends FlowerTurn {
    public TPinkTulip(NeoPlayer player) { super(player); }
    public TPinkTulip(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 5;
    }

    @Override
    public void apply() {
        player.game.allowedClazzes().add(SupernaturalClazz.class);
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.ExtraTurn());
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/flower/TPinkTulip.kt
package net.hectus.neobb.modes.turn.default_game.flower

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.SupernaturalClazz
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class TPinkTulip(data: Block?, cord: Cord?, player: NeoPlayer?) : FlowerTurn(data, cord, player) {
    override val cost: Int = 5

    override fun apply() {
        player!!.game.allowed.add(SupernaturalClazz::class)
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(ExtraTurn())
    }
}
